src/jitlayers.cpp

```diff
@@ -644,6 +649,24 @@ class JLDebuginfoPlugin : public ObjectLinkingLayer::Plugin {
 }
 
 # ifdef LLVM_SHLIB
+#  if JL_LLVM_VERSION >= 140000
+class JLEHFrameRegistrar final : public jitlink::EHFrameRegistrar {
+public:
+    Error registerEHFrames(orc::ExecutorAddrRange EHFrameSection) override {
+        register_eh_frames(
```

Comment:
Looking at this more closely, it seems that only the arguments and unpacking of the arguments have changed between the versions, and the class structure overall is still the same. I wonder if we can simply define two macros, one for the argument and one to unpack it into (de)register_eh_frames which changes between 140000 and below. This would reduce the amount of code that's guarded by the preprocessor here.

Reply:
Thanks for the suggestion. Does this look more clean?